tracer/test/Datadog.Trace.ClrProfiler.Managed.Tests/AutoInstrumentation/AWS/PublishRequestProxy.cs

```diff
@@ -0,0 +1,41 @@
+// <copyright file="PublishRequestProxy.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections;
+using System.Linq;
+using Amazon.SimpleNotificationService.Model;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SNS;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SNSTests
+{
+    public class PublishRequestProxy : IContainsMessageAttributes
+    {
+        private readonly PublishRequest _publishRequest;
+
+        public PublishRequestProxy(PublishRequest publishRequest)
+        {
+            _publishRequest = publishRequest;
+        }
+
+        public IDictionary MessageAttributes
+        {
+            get
+            {
+                // Convert Dictionary<string, MessageAttributeValue> to IDictionary
+                return _publishRequest.MessageAttributes;
+            }
+
+            set
+            {
+                // Convert IDictionary back to Dictionary<string, MessageAttributeValue>
+                _publishRequest.MessageAttributes = value
+                                                   .Cast<DictionaryEntry>()
```

Comment:
I tested locally and this currently throws a InvalidCastException if you try to set MessageAttributes with a `Dictionary<string, MessageAttributeValue>` object. Here's the fix:

```suggestion
                                                   .Cast<KeyValuePair<string, MessageAttributeValue>>()
```

Reply:
Oh this also requires adding the following using statement above:

```csharp
using System.Collections.Generic;
```

Reply:
really odd, I wasn't getting those on my end, the tests were passing in .net 6, 7
<img width="1258" alt="Screenshot 2023-07-20 at 3 58 57 PM" src="https://github.com/DataDog/dd-trace-dotnet/assets/49878080/9340ae93-bdc9-4171-a2fd-ad6b9fcc8e1b">

Reply:
i'm on m1 so maybe you are getting it in a older .net version or i'm missing something else